torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp

```diff
@@ -711,22 +645,25 @@ ReductionParams outerPersistentHeuristic(
     }
   }
 
+  std::cout << "3: " << bdimx << ", " << bdimy << std::endl;
   // Fill bdimy with left over threads
   bdimy = std::min(
       scheduler_utils::safeDiv(max_threads_in_block, bdimx),
       total_reduction_numel);
 
-  // Increase bdimy until we hit the SM partition granularity
-  auto num_warp_per_partition =
-      ceilDiv(bdimx * bdimy, 32 * scheduler_utils::getGpuSmPartitions());
-  while (ceilDiv(
-             bdimx * (bdimy + 1), 32 * scheduler_utils::getGpuSmPartitions()) ==
-         num_warp_per_partition) {
-    bdimy++;
-  }
+  //   // Increase bdimy until we hit the SM partition granularity
+  //   auto num_warp_per_partition =
+  //       ceilDiv(bdimx * bdimy, 32 * scheduler_utils::getGpuSmPartitions());
+  //   while (ceilDiv(
+  //              bdimx * (bdimy + 1), 32 *
+  //              scheduler_utils::getGpuSmPartitions()) ==
+  //          num_warp_per_partition) {
+  //     bdimy++;
+  //   }
 
   bool vectorize = false;
 
+  std::cout << "4: " << bdimx << ", " << bdimy << std::endl;
   // Move unrolling factor into vectorization upto vectorization limit.
   if (vectorize_factor > 1 && iter_unroll_factor > 1) {
```

Comment:
This block needs to be protected because it's setting `iter_unroll_factor` but not checking `max_multi_reduction_factor`. We need to change this block to:
```
  if (vectorize_factor > 1 && iter_unroll_factor > 1) {
    vectorize = true;
    iter_unroll_factor = std::min(
        scheduler_utils::lastPow2(iter_unroll_factor),
        (int64_t)vectorize_factor);
    iter_unroll_factor = std::min(
        iter_unroll_factor,
        scheduler_utils::safeDiv(max_multi_reduction_factor, bdimx));
  }
  ```